src/pages/understand-stacks/running-api-node.md

```diff
@@ -0,0 +1,300 @@
+
+```sh
+docker pull blockstack/stacks-blockchain-api && docker pull blockstack/stacks-blockchain && docker pull postgres:alpine
+docker network create stacks-blockchain > /dev/null 2>&1
+```
+
+Create a directory structure for the service data with the following command:
+
+```sh
+mkdir -p ./stacks-node/{persistent-data/postgres,persistent-data/stacks-blockchain,bns,config} && cd stacks-node
+```
+
+## Step 2: running Postgres
+
+The `postgres:alpine` Docker container can be run with default settings. You must set the password for the user to
+`postgres` with the `POSTGRES_PASSWORD` environment variable. The following command starts the image:
+
+```sh
+docker run -d --rm \
+  --name postgres \
+  --net=stacks-blockchain \
+  -e POSTGRES_PASSWORD=postgres \
+  -v $(pwd)/persistent-data/postgres:/var/lib/postgresql/data \
+  -p 5432:5432
+  postgres:alpine
+```
+
+You can verify the running Postgres instance on port `5432` with the command
+
+```sh
+docker ps --filter name=postgres
+```
+
+## Step 3: running Stacks blockchain API
+
+The `stacks-blockchain-api` image requires several environment variables to be set. To reduce the complexity of the
+run command, you should create a new `.env` file and add the following to it using a text editor:
+
+```dotenv
+NODE_ENV=production
+GIT_TAG=master
+PG_HOST=postgres
+PG_PORT=5432
+PG_USER=postgres
+PG_PASSWORD=postgres
+PG_DATABASE=postgres
+STACKS_CHAIN_ID=0x00000001
+V2_POX_MIN_AMOUNT_USTX=90000000260
+STACKS_CORE_EVENT_PORT=3700
+STACKS_CORE_EVENT_HOST=0.0.0.0
+STACKS_BLOCKCHAIN_API_PORT=3999
+STACKS_BLOCKCHAIN_API_HOST=0.0.0.0
+STACKS_BLOCKCHAIN_API_DB=pg
+STACKS_CORE_RPC_HOST=stacks-blockchain
+STACKS_CORE_RPC_PORT=20443
+BNS_IMPORT_DIR=/bns-data
+```
+
+-> In this example, this guide configured the API to import BNS data with the `BNS_IMPORT_DIR` variable. To disable this
+import, comment the line out by placing a `#` at the beginning of the line. If you leave the BNS import enabled, it may
+take several minutes for the container to start while it imports the data.
+
+The `PG_HOST` and `STACKS_CORE_RPC_HOST` variables define the container names for `postgres` and `stacks-blockchain`.
+You may wish to alter those values if you have named those containers differently than this guide.
+
+Start the `stacks-blockchain-api` image with the following command:
+
+```sh
+docker run -d --rm \
+  --name stacks-blockchain-api \
+  --net=stacks-blockchain \
+  --env-file $(pwd)/.env \
+  -v $(pwd)/bns:/bns-data \
+  -p 3700:3700 \
+  -p 3999:3999 \
+  blockstack/stacks-blockchain-api
+```
+
+You can verify the running `stacks-blockchain-api` container with the command:
+
+```sh
+docker ps --filter name=stacks-blockchain-api
+```
+
+## Step 4: running Stacks blockchain
+
+In order for the API to be functional, the `stacks-blockchain-api` container must have data from a running
+[`stacks-blockchain`][] instance. First create the `./config/Config.toml` file and add the following content to the
```

Comment:
This looks like it should be a link, but the link is missing. 
[`stacks-blockchain`](https://github.com/blockstack/stacks-blockchain)

Reply:
This is a Markdown remote link - I use them to improve readability in the source. You can see the remote definition of the link on line 300.

Reply:
ahh interesting, thanks! downside of checking these changes only in text and not being able to smoke-test how it would look in the prod site